### PR TITLE
"Dispense behavior" options for items

### PR DESCRIPTION
**Pull request by AlsoSomeoneElse**
This PR adds some options to specify the dispense behavior of an item, whether it's dispensed successfully, and what item it becomes after being dispensed. It includes three procedure templates specifically to damage/consume/change the item after being dispensed.

This PR also adds support for itemstack procedure selectors, and procedures with a return value can now pass any amount of dependencies (they caused compilation errors if more than 5 were passed).

### CODE REVIEW

**Comment by Max094Reikeb** [comment]
Looks like you got a problem...
Capture d’écran 2020-11-04 à 20 58 01

File: src/test/java/net/mcreator/integration/TestWorkspaceDataProvider.java [hunk 1]
```diff
@@ -899,6 +899,9 @@ private static GeneratableElement getExampleFor(ModElement modElement, Random ra
 			item.onDroppedByPlayer = new Procedure("procedure9");
 			item.enableMeleeDamage = !_true;
 			item.damageVsEntity = 3;
+			item.hasDispenseBehavior = _true;
+			item.dispenseSuccessCondition = new Procedure("condition1");
+			item.dispenseResultItemstack = new Procedure("itemstack1");
```
Comment (Max094Reikeb) [inline src/test/java/net/mcreator/integration/TestWorkspaceDataProvider.java:904]: There's something wrong with those lines, check my previous comment to see the error.

**Review by Max094Reikeb — CHANGES_REQUESTED** [review]
There's a bug (as said earlier) to be fixed!

**Comment by Defeatomizer** [comment]
>This PR also adds support for itemstack prpcedure selectors

This is actually already supported by MCreator.

**Comment by Max094Reikeb** [comment]
It looks like your problem is still there...

**Review by Max094Reikeb — APPROVED** [review]
Sounds good!

**Review by Goldorion — APPROVED** [review]

**Comment by Goldorion** [comment]
It seems to be ok. Also, will you be able to add this feature for the food mod element as it also extends to the Item class?

**Pull request merged**